### ISSUE TITLE
Docker won't start anymore after windows 1903 update

### ISSUE THREAD

**Issue by MelvinLoos** [docker/for-win#4549]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 9BD69B03-FB45-4ACC-B0FB-C303C5E9A8EA/20190819102053

### Expected behavior
Docker-desktop for windows starts without crashing
### Actual behavior
Docker-desktop for windows crashes when trying to start. It does so when the computer boots and also when I start it manually. It show the following message:

```
Unable to create - The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Generic failure 

   at Docker.Core.Pipe.NamedPipeClient.<TrySendAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Docker.Core.Pipe.NamedPipeClient.Send(String action, Object[] parameters)
   at Docker.Actions.DoStart(SynchronizationContext syncCtx, Boolean showWelcomeWindow, Boolean executeAfterStartCleanup)
   at Docker.Actions.<>c__DisplayClass24_0.<Start>b__0()
   at Docker.ApiServices.TaskQueuing.TaskQueue.<>c__DisplayClass17_0.<.ctor>b__1()
```

### Information

  - Windows Version: Windows 10 Enterprise, version 1903, OS build 18362.295
  - Docker Desktop Version: v2.1.0.1

This al happened when I updated my windows to the latest version, 1903. Before that it started without any problems. I cannot really confirm but I think it has something to do with the communication between docker desktop and hyper-v. I would expect docker to (re)create a network adapter but don't see any from docker. Oh also this is a company laptop and it runs eset so it could also be some security restriction that got more restrictive after the update... any tips are welcome because I just ran out of ideas :(

I have already tried the following:

- Reinstall docker several times (both through choco as through the original installer from docker)
- Reset docker to factory defaults
- Reinstalled hyper-v
- Removed all hyper-v network adapters
- Rebooted several times between all different attempts
- Tried the following tips to fix Virtual Machine Management service of hyper-v https://github.com/docker/for-win/issues/3764
- Ran "MOFCOMP %SYSTEMROOT%\System32\WindowsVirtualization.V2.mof" to rebuild the virtualization module.

### Steps to reproduce the behavior

  1. ... Updated my laptop to latest windows 1903 with docker already installed
  2. ... Try to run docker

Latest logfile:
[log.txt](https://github.com/docker/for-win/files/3515564/log.txt)

**EDIT**: Ok just discovered that when I open the Virtual Switch Manager in hyper-v I see the same message: Generic failure. And looking at the docker logs I see it fail after it tries to create the DockerNAT switch. So it must be related to hyper-v.

**Comment by jpl-jengelke** [comment]
Try my comment near the end of this … https://github.com/docker/for-win/issues/3597

**Comment by MelvinLoos** [comment]
Thank you for the follow up, I really appreciate that! Unfortunately in my case windows messed up my network adapters in the update to 1903 which made it impossible to run any virtualization. The update itself already had trouble going through and only happened after I forced it with powershell so that was already a bad idea in hindsight. Anyway after trying every possible thing I saw no other option then to reset windows (complete OS reinstall). After that everything worked as expected again! So closing the issue.
